Cherry-pick conflicts: features-based-on-v0.8.5.post1 → features-based-on-v0.10.0rc1

CatherineSue opened the pull request:
## 🚧 Cherry-pick Conflicts Need Resolution

This PR contains merge conflicts from cherry-picking commits from `features-based-on-v0.8.5.post1` to `features-based-on-v0.10.0rc1`.

**Source:** `features-based-on-v0.8.5.post1` (commits after tag `v0.8.5.post1`)
**Target:** `features-based-on-v0.10.0rc1` (based on tag `v0.10.0rc1`)

**Commits to cherry-pick:**
- c9bd9f08d4d1d4c865aa292f49e09a8792191e2c
- 236db587734c760ff4ceae0100ea06aa40446e05
- 4dd7951ca53413b2e75caa29db038cc41b82d94f
- 1ddce9a89919f4a5cf889bacf19f5d5612478d7e
- f085ef2be5c7a4998003a748622c555338d6ce32

## 🛠️ How to resolve conflicts:

1. **Checkout this branch:**
  ```bash
  git fetch origin && git checkout cherry-pick-conflicts-features-based-on-v0.10.0rc1
  ```

2. **Resolve conflicts using IDE tools or manually:**
  - **VS Code**: Automatically detects `<<<<<<<`, `=======`, `>>>>>>>` markers and shows merge conflict UI
  - **Other IDEs**: Most modern IDEs detect conflict markers automatically
  - **Manual**: Edit files to remove conflict markers and choose the correct code

3. **Commit the resolution:**
  ```bash
  git add resolved-files
  git commit -m "✅ Resolved: Cherry-pick from features-based-on-v0.8.5.post1 to features-based-on-v0.10.0rc1"
  git push origin cherry-pick-conflicts-features-based-on-v0.10.0rc1
  ```

🎯 **No force push needed!**

upfixer committed:
pick opc-request-id middleware and handle_pydantic_validation_error

CatherineSue commented:
LGTM. Can you attach a screenshot for opc-request-id logs? 
basically add `opc-request-id` in the header like:
```
response = requests.post(
        "http://localhost:8080/v1/chat/completions",
        headers={"Content-Type": "application/json", "opc-request-id": "test01-opc"},
        json=payload,
        stream=True
    )
```

upfixer commented:
![image](https://github.com/user-attachments/assets/a00710ca-cffa-4b07-854a-c248d6fe22b8)

upfixer commented:
![image](https://github.com/user-attachments/assets/44000bd0-b428-4fae-9c96-c66f55879476)

mgoin committed:
[Model] Add tuned triton fused_moe configs for Qwen3Moe (#17328)

Signed-off-by: mgoin <mgoin64@gmail.com>

robertgshaw2-redhat committed:
[BugFix] Fix Memory Leak (#17567)

Signed-off-by: rshaw@neuralmagic.com <robertgshaw2@gmail.com>

LucasWilkinson committed:
[BugFix][Attention] Fix sliding window attention in V1 giving incorrect results (#17574)

Signed-off-by: Lucas Wilkinson <lwilkinson@neuralmagic.com>

key4ng committed:
[Bugfix] Improve JSON extraction in LlamaToolParser (#2)

* [Bugfix] Improve JSON extraction in LlamaToolParser

Signed-off-by: keru <keyang.ru@oracle.com>

* Support Llama 4 multi tool calls use case

Signed-off-by: keru <keyang.ru@oracle.com>

* Fix the regex pattern

Signed-off-by: keru <keyang.ru@oracle.com>

* Resolve comments

Signed-off-by: keru <keyang.ru@oracle.com>

---------

Signed-off-by: keru <keyang.ru@oracle.com>
Co-authored-by: keru <keyang.ru@oracle.com>

Commit:
Add dev container

Signed-off-by: keru <keyang.ru@oracle.com>

Commit:
Add devcontainer.json

Signed-off-by: keru <keyang.ru@oracle.com>

key4ng committed:
Merge pull request #3 from moirai-internal/dev_container

Add dev container Dockerfile and devcontainer.json

github-actions[bot] committed:
🚧 CONFLICTS: Cherry-pick from features-based-on-v0.8.5.post1 to features-based-on-v0.10.0rc1

  ⚠️  This commit contains unresolved merge conflicts in files:
  benchmarks/kernels/benchmark_moe.py tests/v1/core/test_scheduler.py vllm/entrypoints/openai/api_server.py vllm/entrypoints/openai/tool_parsers/llama_tool_parser.py vllm/v1/attention/backends/flash_attn.py vllm/v1/core/sched/scheduler.py

  Commits being cherry-picked:
  c9bd9f08d pick opc-request-id middleware and handle_pydantic_validation_error
236db5877 [Bugfix] Improve JSON extraction in LlamaToolParser (#2)
4dd7951ca Add dev container
1ddce9a89 Add devcontainer.json
f085ef2be Merge pull request #3 from moirai-internal/dev_container

  📝 Edit files to remove conflict markers

CatherineSue commented:
Close this PR as v0.10.0rc1 is not an official tag/release.
